src/main/java/com/google/sps/servlets/ProfileCreationServlet.java

```diff
@@ -38,33 +38,72 @@
    * doPost creates a new user entity and assigns it the property inputed in the user creation form
    */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
+  public void doPost(HttpServletRequest request, HttpServletResponse response)
     throws ServletException, IOException {
-    String username = request.getParameter("username");
-    String bio = request.getParameter("bio");
 
     HttpSession session = request.getSession();
+    if (((int)session.getAttribute("isLoggedIn"))== 1) {
+        editProfile(request, response, session);
+    } else {
+        createProfile(request, response, session);
+    }
+  }
+
+  public void createProfile(
+    HttpServletRequest request,
+    HttpServletResponse response,
+    HttpSession session
+   )  throws IOException {
+    String username = request.getParameter("username");
+    String bio = request.getParameter("bio");
+    String email = (String) session.getAttribute("uregisteredUserEmail");
+ 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
 
     Entity userEntity = new Entity("User");
-    setEnitityAttributes(userEntity, username, bio, session);
+    setEnitityAttributes(userEntity, username, bio, email);
     datastore.put(userEntity);
 
     setSessionAttributes(session, userEntity, username, bio);
 
     response.sendRedirect("/user?id=" + userEntity.getKey().getId());
   }
 
+   public void editProfile(
+    HttpServletRequest request,
+    HttpServletResponse response,
+    HttpSession session
+   )  throws IOException {
+    String username = request.getParameter("username");
+    String bio = request.getParameter("bio");
+    String email = "";
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    long id = (long) session.getAttribute("id");
+    Entity userEntity = null;
+    try {
+      userEntity = getUserById(datastore, id);
+      email = (String) userEntity.getProperty("email");
+    } catch (EntityNotFoundException e) {
+      request.setAttribute("error", 1);
```

Comment:
If you don't need anything else from the lines below, I'd do 

request.setAttribute("error", 1);
response.sendRedirect("/user?id=" + id);
return;

Here.